docs/services/ml/upload-training-script.md

```diff
@@ -0,0 +1,216 @@
+# specifically for getting the label annotations
+def parse_filenames_and_labels_from_json(
+    filename: str, all_labels: ty.List[str]
+) -> ty.Tuple[ty.List[str], ty.List[str]]:
+    """Load and parse JSON file to return image filenames and
+    corresponding labels.
+    Args:
+        filename: JSONLines file containing filenames and labels
+        all_labels: list of all labels present in dataset
+    """
+    image_filenames = []
+    image_labels = []
+
+    with open(filename, "rb") as f:
+        for line in f:
+            json_line = json.loads(line)
+            image_filenames.append(json_line["image_path"])
+
+            annotations = json_line["classification_annotations"]
+            labels = []
+            for annotation in annotations:
+                if annotation["annotation_label"] in all_labels:
+                    labels.append(annotation["annotation_label"])
+            image_labels.append(labels)
+    return image_filenames, image_labels
+
+
+# This is used for parsing the dataset file produced and stored in Viam
+# specifically for getting the bounding box annotations
+def parse_filenames_and_bboxes_from_json(
+    filename: str,
+    all_labels: ty.List[str],
+) -> ty.Tuple[ty.List[str], ty.List[str], ty.List[ty.List[float]]]:
+    """Load and parse JSON file to return image filenames
+    and corresponding labels with bboxes.
+    Args:
+        filename: JSONLines file containing filenames and bboxes
+    """
+    image_filenames = []
+    bbox_labels = []
+    bbox_coords = []
+
+    with open(filename, "rb") as f:
+        for line in f:
+            json_line = json.loads(line)
+            image_filenames.append(json_line["image_path"])
+            annotations = json_line["bounding_box_annotations"]
+            labels = []
+            coords = []
+            for annotation in annotations:
+                if annotation["annotation_label"] in all_labels:
+                    labels.append(annotation["annotation_label"])
+                    # Store coordinates in rel_yxyx format so that
+                    # we can use the keras_cv function
+                    coords.append(
+                        [
+                            annotation["y_min_normalized"],
+                            annotation["x_min_normalized"],
+                            annotation["y_max_normalized"],
+                            annotation["x_max_normalized"],
+                        ]
+                    )
+            bbox_labels.append(labels)
+            bbox_coords.append(coords)
+    return image_filenames, bbox_labels, bbox_coords
+```
+
+{{% /expand%}}
+{{%expand "Step 3: Add logic to produce model artifact" %}}
+
+After reading in the dataset and preparing the data for training, you can add logic to build and compile the model and train the model on the data.
+The code in this part of your script depends on the type of ML training you are doing.
```

Comment:
It would be good to link to the logic from Tahiya's script as an example if it's publicly available

Reply:
done I think